arrows/mvg/algo/initialize_cameras_landmarks.cxx

```diff
@@ -2437,7 +2437,8 @@ ::metadata_centric_keyframe_initialization(
       vector_3d pos_loc;
       rotation_d R_loc;
 
-      if (constraints->get_camera_position_prior_local(fid, pos_loc) &&
+      if (constraints == nullptr &&
```

Comment:
Yes, this is the right thing to do.  Though you could more compactly write `if (constraints &&` since the value of a nullptr is zero which casts to `false` in Boolean.

Reply:
Better to move this check right after cam->clear(); call before "for" loop, since if constraints is nullptr, the loop is useless. Something like this
    `
    cams->clear();
    if (!constraints)
    {
      return false;
    }
    `
it can be returned immediately because anway after it goes out of if-else block, `cams->size() < 2` condition is being checked, which means it will always return false there, so we can leave the function immediately. Otherwise `for` loop needs to be in block `if (constraints) { for ...`